libcaf_core/src/detail/rfc3629.cpp

```diff
@@ -72,18 +90,25 @@ bool validate_rfc3629(const std::byte* first, const std::byte* last) {
         return false;
       continue;
     }
-    // 1111'0bxx: 4-byte sequence.
+    // 1111'0xxx: 4-byte sequence.
     if (head<5>(x) == 0b1111'0000_b) {
+      uint64_t code_point = std::to_integer<uint64_t>(tail<3>(x)) << 18;
```

Comment:
I don't think we need to compute this here. We won't read this value until line 101. Why not move it below and merge with the first `|=`?